src/test/java/net/snowflake/ingest/streaming/internal/datatypes/AbstractDataTypeTest.java

```diff
@@ -112,6 +122,21 @@ protected String createTable(String dataType) throws SQLException {
             String.format(
                 "create or replace table %s (%s string, %s %s)",
                 tableName, SOURCE_COLUMN_NAME, VALUE_COLUMN_NAME, dataType));
+
+    return tableName;
+  }
+
+  protected String createIcebergTable(String dataType) throws SQLException {
```

Comment:
also add an argument for `boolean nullable` and we can make sure the sdk does proper nullability checking for columns marked non-nullable.

Reply:
Added nullable test.

Reply:
I see nullable was removed, why so?

Reply:
Removed and use `<datatype> not null` in datatype testing. By doing this we can set nullable on any sub-columns on structured data type.